crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -1400,6 +1400,7 @@ pub struct SuiGasData {
 #[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, PartialEq, Eq)]
 #[serde(rename = "TransactionData", rename_all = "camelCase")]
 pub struct SuiTransactionData {
+    pub message_version: u64,
```

Comment:
Agreed with @lxfind's assessment here, I think it would be harder to evolve if there's a big change in `TransactionData` and we have to stick to adding/deprecating fields in this struct rather than having a new enum variant. It would be really great if we could somehow make the internally tagged enum serialization work.

That said, I'm okay with landing it as it is as this looks good from an API perspective(but not from a backward compatibility evolution perspective, which can be done in a way that does not affect API).

Reply:
@666lcz we can probably use the tagged serialization for rpc types, since those don't have to work with the crypto primitives. let me try.

Reply:
@666lcz PTAL - I made the change for SuiTransactionData. Let me know if that's what you had in mind, and I'll do the same for SuiTransactionEffects

Reply:
This is amazing!! Thanks for looking into this!